### PR TITLE
:children_crossing: handle errors when retrieving configuration

### PR DESCRIPTION
**Pull request by hgwood**
When an error occurred while retrieving configuration, the client would show a infinite loading screen. This change makes a crude error appear instead.

There's an optional refactoring commit in there, you might no want it.

### CODE REVIEW

**Comment by Errorname** [comment]
Good PR!

However, `getConfiguration` is not a middleware anymore and this could conflict with its use in `server/src/index.js`

**Comment by hgwood** [comment]
Good catch, I thought there was a good reason that is used callback-style :) I'll revert the refactoring.

**Comment by Errorname** [comment]
Maybe one day I'll convert the server codebase to Typescript to help refactoring :)

File: server/src/middlewares/configuration.js [hunk 1]
```diff
@@ -6,8 +6,12 @@ const getConfiguration = async (multiTenant, req, next) => {
     return
   }
 
-  await refreshConfiguration(tenant)
-  next()
+  try {
```
Comment (Errorname) [inline server/src/middlewares/configuration.js]: You could do:
```js
await refrechConfiguration(tenant).catch(next)
next()
```

File: src/services/configuration.js [hunk 2]
```diff
@@ -39,19 +45,26 @@ class ConfigurationProvider extends Component {
...
+        .then(() => this.setState({ loaded: true }))
+        .catch(err => this.setState({ error: err }))
     } else {
       // Else, load the configuration in the background anyway
-      configuration.load()
+      configuration.load().catch(err => console.warn('Configuration could not be refreshed:', err))
     }
   }
   render() {
+    if (this.state.error) {
```
Comment (Errorname) [inline src/services/configuration.js]: 👍

**Comment by hgwood** [comment]
> Maybe one day I'll convert the server codebase to Typescript to help refactoring :)

Maybe I tried just that this week-end 🙄

**Comment by Errorname** [comment]
Ahah, how did it go? 😬

File: server/src/middlewares/configuration.js (hunk 1 above)
Comment (hgwood) [reply server/src/middlewares/configuration.js]: Done

**Comment by hgwood** [comment]
> Ahah, how did it go? 😬

I manage to get the question resolvers typed. However it looks like the prisma binding types are really limited. Prisma client should work better.

**Comment by Errorname** [comment]
Indeed, I think we should use Prisma client with generated TS. That would work way better

**Comment by hgwood** [comment]
Agreed. I'll experiment. Is this PR ready to merge?

**Comment by Errorname** [comment]
I haven't seen your last commit ?

**Comment by hgwood** [comment]
Oh, I guess I forgot to push 😅

**Comment by hgwood** [comment]
Pushed.

**Pull request merged**